src/attributes/components/AttributeDetails/NumericUnits.tsx

```diff
@@ -28,46 +37,68 @@ const useStyles = makeStyles(
   { name: "NumericUnits" }
 );
 
-interface NumericUnitsProps {
-  data: AttributePageFormData;
+interface UnitData {
+  unit: MeasurementUnitsEnum;
+  system: UnitSystem;
+  type: UnitType;
+}
```

Comment:
I see that the state uses `Partial<UnitData>` while initialData is defined `as UnitData` still having the chance for undefined object members. Also I think we should avoid using `as` assertion if possible to let compiler safely infer type. So, why not to simply make all properties optional?
```suggestion
interface UnitData {
  unit?: MeasurementUnitsEnum;
  system?: UnitSystem;
  type?: UnitType;
}
```